java_cucumber/src/test/java/java_cucumber/Stepdefs.java

```diff
@@ -930,4 +943,290 @@ public void newAccInfo() throws ApiException, NoSuchAlgorithmException, com.algo
+    public void i_create_a_freeze_transaction_targeting_the_second_account() throws NoSuchAlgorithmException, ApiException, com.algorand.algosdk.kmd.client.ApiException {
+        this.renewHandle(); // to avoid handle expired error
+        TransactionParams params = acl.transactionParams();
+        lastRound = params.getLastRound();
+
+        Transaction tx = Transaction.createAssetFreezeTransaction(
+                new Address(this.accounts.get(0)), // transaction sender
+                new Address(this.accounts.get(1)), // revoked from
+                true, // freeze state
+                acl.transactionParams().getFee(), // transaction fee
+                acl.transactionParams().getLastRound(), // first valid round 
+                acl.transactionParams().getLastRound().add(BigInteger.valueOf(1000)), // last valid round
+                this.note,
+                new Digest(acl.transactionParams().getGenesishashb64()),
+                this.assetID);
+        Account.setFeeByFeePerByte(tx, tx.fee);
+        this.txn = tx;
+        this.pk = new Address(this.accounts.get(0));
+    }
+
+    @When("I create a transaction transferring {int} assets from a second account to creator")
+    public void i_create_a_transaction_transferring_assets_from_a_second_account_to_creator(Integer int1) throws ApiException, NoSuchAlgorithmException {
+        TransactionParams params = acl.transactionParams();
+        lastRound = params.getLastRound();
+
+        Transaction tx = Transaction.createAssetTransferTransaction(
+                new Address(this.accounts.get(1)),
+                new Address(this.creator), // sender source address                
+                new Address(),
+                BigInteger.valueOf(int1),
+                acl.transactionParams().getFee(), // transaction fee
+                acl.transactionParams().getLastRound(), // first valid round 
+                acl.transactionParams().getLastRound().add(BigInteger.valueOf(1000)), // last valid round
+                this.note,
+                acl.transactionParams().getGenesisID(),
+                new Digest(acl.transactionParams().getGenesishashb64()),
+                this.assetID);
+        Account.setFeeByFeePerByte(tx, tx.fee);
+        this.txn = tx;
+        this.pk = new Address(this.accounts.get(1));
+    }
+
+    @When("I create an un-freeze transaction targeting the second account")
+    public void i_create_an_un_freeze_transaction_targeting_the_second_account() throws ApiException, NoSuchAlgorithmException, com.algorand.algosdk.kmd.client.ApiException  {
+        this.renewHandle(); // to avoid handle expired error
+        TransactionParams params = acl.transactionParams();
+        lastRound = params.getLastRound();
+
+        Transaction tx = Transaction.createAssetFreezeTransaction(
+                new Address(this.accounts.get(0)), // transaction sender
+                new Address(this.accounts.get(1)), // revoked from
```

Comment:
I don't think this transaction is revoking any assets. Is this the account being frozen?

Reply:
the comment should say "account to freeze" instead of "revoked from"